### DISCUSSION TITLE
Messages filtering with header vs value

### DISCUSSION THREAD

**Discussion by Zeeeeta** [kafbat/kafka-ui#467] [Q&A]
Hi,

I have done testing on a large data set, where I have a same attribute in both the header and the value.
There are a few attributes in header, and roughly 50+ attributes in value for each message.

However, when I do a polling testing, I found that the Elapsed Time and Bytes Consumed are almost the same.

```
By Value
1064466 ms
31 MB
27193 messages consumed
```
![kafka_filter_by_value](https://github.com/kafbat/kafka-ui/assets/26813698/4fc32d54-2c46-4da5-bc58-b3db4111446d)
```
By Header
1064422 ms
31 MB
27199 messages consumed
```
![kafka_filter_by_header](https://github.com/kafbat/kafka-ui/assets/26813698/f4de109e-259f-41e7-8efc-e4832fb40dc8)

Does Kafbat not able to filter by header instead of consuming entire message to improve polling performance?

Thanks.

**Comment by Haarolean** [comment]
In order to apply the filtering we need to consume the messages, that's why adjusting the filter won't affect the mentioned metrics

**Reply by Zeeeeta** [reply]
> In order to apply the filtering we need to consume the messages, that's why adjusting the filter won't affect the mentioned metrics

Thanks for replying.
Since I have read some best practices where header can help with filtering.

In that case, since both my header and value are plain json, and the consumer application has to consume the entire message anyway, there is no change in performance.

Maybe header is only useful for performance when I have body compression or even encryption?
